core/src/main/java/org/pentaho/di/core/vfs/KettleVFS.java

```diff
@@ -522,4 +522,16 @@ public static void closeEmbeddedFileSystem( String embeddedMetastoreKey ) {
     }
   }
 
+  /**
+   * Reboot default file system manager and by consequence shutting down any pending SFTP connection
+   */
+  public static void rebootDefaultFileSystem() {
+    FileSystemManager fsManager = getInstance().getFileSystemManager();
+
+    if ( fsManager != null ) {
+      ((DefaultFileSystemManager) fsManager).close();
+      kettleVFS = new KettleVFS();
```

Comment:
We cannot do this in this way. PDI is a multi-threaded application, and used in the Pentaho server. Imagine 15 simultaneous threads working with the filesystem instance as one of the threads swaps out the filesystem manager. We need to look at this more carefully. Likely need to use a ThreadLocal or similar so that each thread gets it's own kettleVFS. Ultimately, cannot have one thread yank the rug out from under another thread.